trl/trainer/ppo_trainer.py

```diff
@@ -252,6 +255,17 @@ def _step_safety_checker(
                     f"Batch size ({batch_size}) does not match number of examples - but got {len(tensor_list)} for: {name}"
                 )
 
+            # set scores on the correct device
+            if name == "scores":
+                scores = [score.to(self.accelerator.device) for score in scores]
```

Comment:
I think we can do the same for queries and values as well. although they should be you don't know what the user might do before passing them to `step` and then the behaviour is consistent for all inputs. 

Do you know what happens when the tensor is already on device? Will it copy it again or do nothing?

Reply:
Makes sense, in PT it should do nothing if they are on the same device so this should be cost-free

Reply:
Should be now handled in [b32bfbf](https://github.com/lvwerra/trl/pull/76/commits/b32bfbffaa16abde73e12ba838c311dcfbbd175b)